### DISCUSSION TITLE
Notes from a Small Functional Island

### DISCUSSION THREAD

**Discussion by louthy** [louthy/language-ext#1300] [Announcements]
Those who have been following the repo will know that I am building up to a really massive version 5 release. I am looking at the release notes, that I have been trying to keep up to date as I go (so that I don't forget anything), and they're already up to nearly 1000 lines of text. And, although there's quite a lot of content [in the release notes], there's also 100s of lines of just headings (waiting for the content)! So, the final release notes are going to be huge and a lot to take in.

> Version 5 of language-ext is about as close to a rewrite as it's ever going to get. So there's a lot to cover.

Rather than have a big surprise on day 1 of the release, I figured I'd start a blog to try and prepare the ground and whet the appetite somewhat – and also terrify you enough that you upgrade with caution!

Here's the link: https://paullouth.com/higher-kinds-in-c-with-language-ext/

The first article is about Higher Kinds in C# -- go check it out and subscribe.  All feedback welcome!

Paul

**Comment by josmithua** [comment]
You my friend, are a higher kind. Thanks for sharing your amazing work!

**Comment by louthy** [comment]
Part 2 of ['Higher Kinds in C# with language-ext'](https://paullouth.com/higher-kinds-in-c-with-language-ext-part-2/) is now available.

For those who have signed up, thank you!  I'm still fighting Mailgun to try and get the automated emails working -- hopefully it'll be sorted soon!

**Comment by louthy** [comment]
Part 3 of [Higher Kinds in C#](https://paullouth.com/higher-kinds-in-c-with-language-ext-part-3-foldables/) is now online.  This article covers foldables -- a completely new trait for Language-Ext!

**Comment by louthy** [comment]
For those who haven't yet seen, [part 4 of Higher Kinds in C# is available now](https://paullouth.com/higher-kinds-in-c-with-language-ext-part-4-applicatives/). Next up is applicatives!

**Comment by louthy** [comment]
[Part 5 of Higher Kinds in C# is online now ](https://paullouth.com/higher-kinds-in-c-with-language-ext-part-5-validation/)-- this episode: Validation.

**Comment by louthy** [comment]
Part 6 of my Higher Kinds in C# series [is now available](https://paullouth.com/higher-kinds-in-csharp-part6-traversables/).  In this episode dig into one of the most powerful types in all of functional-programming: Traversable.

**Comment by louthy** [comment]
Part 7 of my Higher Kinds in C# is [now available](https://paullouth.com/higher-kinds-in-csharp-with-language-ext-part-7-monads/).  And yes, it's MONADS! :+1:

**Comment by louthy** [comment]
Part 9 of my Higher Kinds in C# is now available and its [monad transformers!](https://paullouth.com/higher-kinds-in-csharp-with-language-ext-part-9-monad-transformers/) 👍

I realise I also forgot to update this thread with [Part 8](https://paullouth.com/higher-kinds-in-csharp-with-language-ext-part-8-monads-continued/).

**Comment by louthy** [comment]
Part 10 of my Higher Kinds in C# series is now available.  [Continuing the monad transformer story with `ReaderT`](https://paullouth.com/higher-kinds-in-csharp-with-language-ext-part-10-readert-monad-transformer/).

Enjoy :+1:

**Comment by louthy** [comment]
Part 11 of my Higher Kinds in C# series [is now available](https://paullouth.com/higher-kinds-in-csharp-with-language-ext-part-11-statet-monad-transformer/). Continuing the monad-transformers story with `StateT` and a full game implementation of pontoon/21/vingt-un!
